content/blog/2019/mesh-federation/index.md

```diff
@@ -0,0 +1,1816 @@
+---
+title: Mesh Federation
+subtitle: Connect multiple service meshes ad hoc, with limited service exposure and strict cross-cluster access control
+description: Connect multiple service meshes ad hoc, with limited service exposure and strict cross-cluster access control.
+publishdate: 2019-08-12
+attribution: Vadim Eisenberg (IBM)
+keywords: [traffic-management,multicluster,security,gateway,tls]
+---
+
+Managing applications across multiple Kubernetes clusters is currently one of the hottest topics in Kubernetes and Istio
+communities. You may need to split your applications between multiple clusters for different reasons, for example:
+
+-  scalability limits of a single cluster
+-  separation of production/dev/test/staging environments
+-  different security and compliance requirements for different parts of the application
+-  running parts of the application on multiple public clouds, and on a private cloud
+
+Currently there are [three patterns](/docs/setup/kubernetes/install/multicluster/) in Istio to create a
+[multicluster service mesh](/docs/concepts/multicluster-deployments/#multicluster-service-mesh).
+A _multicluster service mesh_ is a single _logical_ service mesh, spread among multiple Kubernetes clusters.
+
+Such a service mesh has the following characteristics:
+
+- **uniform identical naming** of namespaces and services. The `withdraw` service in the `accounts` namespace in one
+cluster has the same functionality and API as the `withdraw` service in the `accounts` namespace in all other clusters
+in the mesh.
+- **expose-all by default**. All the services are exposed by default to all the clusters in the mesh.
+- **common identity and common trust** are established between all the clusters in the mesh.
+
+The three patterns differ by the following aspects:
+
+- The clusters are on a **single network** or on **multiple networks**. Single network means that the pods in one
```

Comment:
This section makes me feel like the reader should first be introduced to the concepts of Istio topologies. I am currently working with the docs team on a first cut of an Istio Topologies concept doc which should be out for review in PR form shortly. I think that doc needs to land on istio.io before this does.

Reply:
@nmittler please note that I propose this blog post to preliminary.istio.io, not to merge it into istio.io yet. Once the terminology changes, I am OK with changing the terms in this blog post.

Reply:
That doesn't really help, since preliminary.istio.io is just a staging area for istio.io. If this lands first, then we're under the gun to make sure my PR is also merged and that we retroactively fix this blog before we cut a release. Not a good workflow.

Reply:
I think I agree with @nmittler that there is too much talk about multicluster mesh in this blog. The blog doesn't really need to talk about the existing documented multicluster patterns, and how they differ, but should just minimally explain what's different about mesh federation, and then follow through with the example.

The concepts doc should explain all the details about multicluster and more details about mesh federation too.

That said, I do think we need to update the concepts doc asap (for 1.3) so even if this blog tries to be vague about the multicluster details, it will probably need to be at least minimally updated when the concepts update lands, but that's not a big problem if the blog stays high level enough.

@nmittler What's the ETA for your concepts PR? Is it complimentary or a replacement of https://github.com/istio/istio.io/pull/4725

Reply:
Already handed my draft of the docs to our docs team who will be turning it into a PR shortly.  It's targeted for 1.3.

Reply:
@nmittler @rcaballeromx had indicated the draft PRs are targeted for 14...